modules/tftp/server.rb

```diff
@@ -90,21 +103,114 @@ def pxe_default
+          logger.debug "TFTP: Directory '#{bootloader_path}' does not exist."
+          return
+        end
+      end
+
+      bootloader_path
+    end
+
+    def create_bootloader_universe_symlinks(bootloader_path, pxeconfig_dir_mac)
+      symlinks = []
+      Dir.glob(File.join(bootloader_path, '*.efi')).each do |source_file|
+        symlinks << { source: Pathname.new(source_file), symlink: Pathname.new(File.join(pxeconfig_dir_mac, File.basename(source_file))) }
+      end
+      create_symlinks(symlinks)
+    end
+
+    def create_default_symlinks(bootfilename_efi, pxeconfig_dir_mac)
+      pxeconfig_dir = pxeconfig_dir()
+      symlinks = [
+        { source: Pathname.new(File.join(pxeconfig_dir, "grub#{bootfilename_efi}.efi")), symlink: Pathname.new(File.join(pxeconfig_dir_mac, "boot.efi")) },
+        { source: Pathname.new(File.join(pxeconfig_dir, "grub#{bootfilename_efi}.efi")), symlink: Pathname.new(File.join(pxeconfig_dir_mac, "grub#{bootfilename_efi}.efi")) },
+        { source: Pathname.new(File.join(pxeconfig_dir, "shim#{bootfilename_efi}.efi")), symlink: Pathname.new(File.join(pxeconfig_dir_mac, "boot-sb.efi")) },
+        { source: Pathname.new(File.join(pxeconfig_dir, "shim#{bootfilename_efi}.efi")), symlink: Pathname.new(File.join(pxeconfig_dir_mac, "shim#{bootfilename_efi}.efi")) },
+      ]
+      create_symlinks(symlinks)
+    end
+
+    def create_symlinks(symlinks)
+      symlinks.each do |link|
+        relative_source_path = link[:source].relative_path_from(link[:symlink].parent)
+
+        logger.debug "TFTP: Creating relative symlink: #{link[:symlink]} -> #{relative_source_path}"
+        FileUtils.ln_s(relative_source_path.to_s, link[:symlink].to_s, force: true)
+      end
+    end
+
+    # Configures bootloader files for a host in its host-config directory
+    #
+    # @param mac [String] The MAC address of the host
+    # @param os [String] The lowercase name of the operating system of the host
+    # @param release [String] The major and minor version of the operating system of the host
+    # @param arch [String] The architecture of the operating system of the host
+    # @param bootfilename_efi [String] The architecture specific boot filename suffix
+    def setup_bootloader(mac:, os:, release:, arch:, bootfilename_efi:)
+      pxeconfig_dir_mac = pxeconfig_dir(mac)
+
+      logger.debug "TFTP: Deploying host specific bootloader files to '#{pxeconfig_dir_mac}'."
+
+      FileUtils.mkdir_p(pxeconfig_dir_mac)
+      FileUtils.rm_f(Dir.glob("#{pxeconfig_dir_mac}/*.efi"))
+
+      bootloader_path = bootloader_path(os, release, arch)
+
+      if bootloader_path
+        logger.debug "TFTP: Creating symlinks from bootloader universe."
+        create_bootloader_universe_symlinks(bootloader_path, pxeconfig_dir_mac)
+      else
+        logger.debug "TFTP: Creating symlinks from default bootloader files."
+        create_default_symlinks(bootfilename_efi, pxeconfig_dir_mac)
+      end
```

Comment:
I think the terms `bootloader universe` or `default bootloader files` might not be clear to users reading these logs. It would be better to either remove these messages or include the actual directory paths for better context.